City-of-Helsinki/drupal-hdbt
# UHF-4235 Fix hds buttons

# Fix hds buttons (part of whitelist links) [UHF-4235](https://helsinkisolutionoffice.atlassian.net/browse/UHF-4235)
As a part of link work we noticed that the hds button colors were behaving badly. This is a sub PR to the whitelist PR trying to fix that problem on a bigger scale.

## What was done
* Simplified theme handling in `<body>` element
* Copy hds button styles into hdbt for refactoring
* Removed unused hds button themes
  * small buttons
  * brand colored buttons
* Convert buttons to use `outline` instead of `::after` for focus indicators
* As we're using only transparent, black and white colors on buttons, simplified color overriding of buttons with mixins
* Cleaned button scss code from several places
  * eu-cookie compliance banner
  * eu-cookie compliance page
  * embedded warning dialogs like youtube, maps
  * wysiwyg areas
  * hero
  * sidebar
  * service lists
  * unit search
* Added missing icon "layers" as it's used in admin theme.

## How to install
* Make sure your instance is up and running on latest dev branch.
    * `git pull origin dev`
    * `make fresh`
* Update HDBT_Admin theme
    * `composer require drupal/hdbt_admin:dev-UHF-4235_fix_hds_buttons`
* Update the HDBT theme
    * `composer require drupal/hdbt:dev-UHF-4235_fix_hds_buttons`
* Make sure that helfi_platform_config is in 4235_whitelist_internal_links branch
* Run `make drush-cr`


## How to test
* Check that buttons behave nicely in all forms.
  * All different button instances
    * eu-cookie compliance banner
    * eu-cookie compliance page
    * embedded warning dialogs like youtube, maps
    * wysiwyg areas
    * hero
    * sidebar
    * service lists
    * unit search
    * anything missing?
  * All button variants should work
    * Primary
    * Secondary
    * Supplementary
  * They should have correct colors in all states
    * Default
    * Hover
    * Focus
    * Hover + Focus
    * Disabled
    * Active (basically same as focus, until we get any designs from UX team) (This should be tested on safari as it's the only browser that does not focus button on click)
* Check that theme switching works as it used to
  * Different selectors
    * Sitewide
    * Page specific
  * Backgrounds should work
    * Buttons should work on backgrounds, we only have two variants of all buttons, black and white. Based on background color the button color should switch to one of these to maintain proper contrast.
* Check that layers icon is added properly to site default icons


## Other PRs
* https://github.com/City-of-Helsinki/drupal-hdbt-admin/pull/118 To handle hdbt_admin theme colors
